game-core/src/test/java/games/strategy/triplea/delegate/battle/steps/FirstStrikeStepOrderTest.java

```diff
@@ -0,0 +1,392 @@
+package games.strategy.triplea.delegate.battle.steps;
+
+import static games.strategy.triplea.delegate.battle.FakeBattleState.givenBattleStateBuilder;
+import static games.strategy.triplea.delegate.battle.steps.BattleStepsTest.givenAnyUnit;
+import static games.strategy.triplea.delegate.battle.steps.BattleStepsTest.givenUnitDestroyer;
+import static games.strategy.triplea.delegate.battle.steps.BattleStepsTest.givenUnitFirstStrike;
+import static games.strategy.triplea.delegate.battle.steps.FirstStrikeStepOrder.DEFENDER_NO_SNEAK_ATTACK;
+import static games.strategy.triplea.delegate.battle.steps.FirstStrikeStepOrder.DEFENDER_NO_SNEAK_ATTACK_BUT_BEFORE_STANDARD_ATTACK;
+import static games.strategy.triplea.delegate.battle.steps.FirstStrikeStepOrder.DEFENDER_SNEAK_ATTACK;
+import static games.strategy.triplea.delegate.battle.steps.FirstStrikeStepOrder.DEFENDER_SNEAK_ATTACK_WITH_OPPOSING_FIRST_STRIKE;
+import static games.strategy.triplea.delegate.battle.steps.FirstStrikeStepOrder.OFFENDER_NO_SNEAK_ATTACK;
+import static games.strategy.triplea.delegate.battle.steps.FirstStrikeStepOrder.OFFENDER_SNEAK_ATTACK;
+import static games.strategy.triplea.delegate.battle.steps.FirstStrikeStepOrder.OFFENDER_SNEAK_ATTACK_WITH_OPPOSING_FIRST_STRIKE;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.hasSize;
+
+import games.strategy.engine.data.Unit;
+import games.strategy.triplea.delegate.battle.BattleState;
+import games.strategy.triplea.delegate.battle.steps.retreat.OffensiveSubsRetreatTest.MockGameData;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+class FirstStrikeStepOrderTest {
+
+  enum BattleStateVariation {
+    HAS_ATTACKING_FIRST_STRIKE,
+    HAS_DEFENDING_FIRST_STRIKE,
+    HAS_ATTACKING_DESTROYER,
+    HAS_DEFENDING_DESTROYER,
+    HAS_WW2V2,
+    HAS_DEFENDING_SUBS_SNEAK_ATTACK
+  }
+
+  BattleState givenBattleState(final List<BattleStateVariation> parameters) {
```

Comment:
Codacy found an issue: [The method 'givenBattleState(List)' has an NPath complexity of 288, current threshold is 200](https://app.codacy.com/gh/triplea-game/triplea/pullRequest?prid=5803017)